src/core/lib/transport/metadata.c

```diff
@@ -504,7 +507,8 @@ static void grow_mdtab(mdtab_shard *shard) {
 }
 
 static void rehash_mdtab(mdtab_shard *shard) {
-  if (shard->free > shard->capacity / 4) {
+  if ((size_t)gpr_atm_no_barrier_load(&shard->free_estimate) >
```

Comment:
Yes, this needs to be ssize_t. (Given that the underlying gpr_atm is signed, everything else is already broken if the capacity doesn't fit in a ssize_t.)